booster/library/Booster/Pattern/ApplyEquations.hs

```diff
@@ -413,14 +438,15 @@ iterateEquations ::
     Term ->
     EquationT io Term
 iterateEquations direction preference startTerm = do
-    logOther (LevelOther "Simplify") $ "Evaluating " <> renderText (pretty startTerm)
     result <- pushRecursion startTerm >>= checkCounter >> go startTerm <* popRecursion
-    logOther (LevelOther "Simplify") $ "Result: " <> renderText (pretty result)
+    when (startTerm /= result) $ withContext "success" $ withTermContext result $ pure ()
     pure result
   where
     checkCounter counter = do
         config <- getConfig
-        when (counter > config.maxRecursion) $
+        when (counter > config.maxRecursion) $ do
+            withContext "abort" $ logMessage ("Recursion limit exceeded" :: Text)
+            logWarnNS "booster" "Recursion limit exceeded"
```

Comment:
:+1: Good to have this as a new warning.
When I looked at the CI results for this PR I noticed that the new test derived from Petar's issue is emitting [a large number of those `Recursion limit exceeded`](https://github.com/runtimeverification/haskell-backend/actions/runs/8851119026/job/24307006419?pr=3826#step:9:2582)
We could point the user to the `--equation-max-recursion` parameter (assuming the name is stable).